src/main/java/org/apache/sling/discovery/oak/SlingIdCleanupTask.java

```diff
@@ -0,0 +1,484 @@
+        logger.info("run: slingId cleanup done.");
+    }
+
+    /**
+     * Do the actual cleanup of garbage slingIds and report back with true if there
+     * might be more or false if we're at the end.
+     * 
+     * @return true if there might be more garbage or false if we're at the end
+     */
+    boolean cleanup() {
+        logger.debug("cleanup: start");
+
+        final ResourceResolverFactory localFactory = resourceResolverFactory;
+        final Config localConfig = config;
+        if (localFactory == null || localConfig == null) {
+            logger.warn("cleanup: cannot cleanup due to rrf={} or c={}", localFactory,
+                    localConfig);
+            return true;
+        }
+        final TopologyView localCurrentView = currentView;
+        if (localCurrentView == null || !localCurrentView.isCurrent()) {
+            logger.info("cleanup : cannot cleanup as topology recently changed : {}",
+                    localCurrentView);
+            return true;
+        }
+        final Set<String> activeSlingIds = new HashSet<>();
+        for (InstanceDescription id : localCurrentView.getLocalInstance().getClusterView()
+                .getInstances()) {
+            activeSlingIds.add(id.getSlingId());
+        }
+
+        ResourceResolver resolver = null;
+        try {
+            resolver = localFactory.getServiceResourceResolver(null);
+
+            final Resource clusterInstances = ResourceHelper.getOrCreateResource(resolver,
+                    localConfig.getClusterInstancesPath());
+            final Resource idMap = ResourceHelper.getOrCreateResource(resolver,
+                    localConfig.getIdMapPath());
+            final Resource syncTokens = ResourceHelper.getOrCreateResource(resolver,
+                    localConfig.getSyncTokenPath());
+            resolver.revert();
+            resolver.refresh();
+
+            final ValueMap idMapMap = idMap.adaptTo(ValueMap.class);
+            final ModifiableValueMap syncTokenMap = syncTokens
+                    .adaptTo(ModifiableValueMap.class);
+            final Calendar now = Calendar.getInstance();
+            int removed = 0;
+            boolean mightHaveMore = false;
+            int localBatchSize = batchSize;
+            long localMinCreationAgeMillis = minCreationAgeMillis;
+            for (Resource resource : clusterInstances.getChildren()) {
+                if (!hasTopology || currentView != localCurrentView
+                        || !localCurrentView.isCurrent()) {
+                    // we got interrupted during cleanup
+                    // let's not commit at all then
+                    logger.info(
+                            "cleanup : topology changing during cleanup - not committing this time - stopping for now.");
+                    return true;
+                }
+                final String slingId = resource.getName();
+                logger.info("cleanup : handling slingId = {}", slingId);
```

Comment:
please downgrade the log level:
```suggestion
                logger.trace("cleanup : handling slingId = {}", slingId);
```

Reply:
lowered to debug already